directory_md/action.yml

```diff
@@ -27,17 +27,24 @@ runs:
       uses: actions/setup-python@v4
       with:
         python-version: '3.10' # or whatever version you support
-    - name: Setup Git configurations
-      shell: bash
-      run: |
-        git config --global user.name github-actions[bot]
-        git config --global user.email 'github-actions@users.noreply.github.com'
     - name: Running the formatter
       shell: bash
       run: |
         python build_directory_md.py ${{ inputs.language }} ${{ inputs.working-directory }} ${{ inputs.filetypes }} ${{ inputs.ignored-directories }} ${{ inputs.ignore-folders-children }} > DIRECTORY.md
-    - name: Committing changes
+
+        git pull || true
+    - name: Committing and pushing changes
+      uses: stefanzweifel/git-auto-commit-action@v4
+      id: commit-push
+      with:
+        commit_message: "docs: updating `DIRECTORY.md`"
+        branch: "directory-${{ github.sha }}"
+        create_branch: true
+    - name: Creating and merging the PR
       shell: bash
+      if: steps.commit-push.outputs.changes_detected == 'true'
       run: |
-        git commit -m "chore: update `DIRECTORY.md`" DIRECTORY.md || true
-        git push origin HEAD:$GITHUB_REF || true
+        gh pr create --base ${GITHUB_REF##*/} --head directory-${{ github.sha }} --title 'docs: updating `DIRECTORY.md`' --body '`DIRECTORY.md` has been updated (see the diff. for changes).'
+        gh pr merge --admin --merge --subject 'docs: updating `DIRECTORY.md`' --delete-branch
+      env:
+        GH_TOKEN: ${{ github.token }}
```

Comment:
I think it might be best to take this out to its own action 🤔 - I remember I did something similar in a different repository.

Reply:
Sure sure! So, should we use the current method for now, and later on create a PR to use that workflow?